### PR TITLE
add relevant description to tqdm in examples

### LINKED ISSUE #11797: [examples] add desc to `dataset.map` to improve tqdm bars
Once https://github.com/huggingface/datasets/pull/2374 is merged we should deploy this feature in our examples.

Currently we get a bunch of bars that are absolutely meaningless and hard to understand what they do. See also: https://github.com/huggingface/datasets/issues/2330

The only issue is how to depend on `datasets` dev versions, might have to wait for a new `datasets` release to be able to merge such PR.

### PR DESCRIPTION
**Pull request by bhavitvyamalik**
# What does this PR do?
Fixes #11797. Added description to `dataset.map` to improve tqdm bars. This would tell the user what's being processed. As of now I've only targeted `text-classification`, please let me know if it looks good or if any changes are required so that I can update the rest of them as well.

I was also thinking we should have `check_min_version` for `datasets` as well as this won't work in dataset versions prior to 1.7.0.

## Who can review?
@stas00

### CODE REVIEW

File: examples/pytorch/text-classification/run_glue.py
Comment (stas00) [inline examples/pytorch/text-classification/run_glue.py:397]: Does it work for you? For me with latest `datasets` master it doesn't know of `desc`.

Here is to verify in isolation:

```
PYTHONPATH=src python -c 'from datasets import load_dataset; ds=load_dataset("glue", "cola"); ds.map(desc="")'
Reusing dataset glue (/home/stas/.cache/huggingface/datasets/glue/cola/1.0.0/dacbe3125aa31d7f70367a07a8a9e72a5a0bfeb5fc42e75c9db75b96da6053ad)
Traceback (most recent call last):
  File "<string>", line 1, in <module>
TypeError: map() got an unexpected keyword argument 'desc'
```

**Comment by stas00** [comment]
> I was also thinking we should have `check_min_version` for `datasets` as well as this won't work in dataset versions prior to 1.7.0.

That and also we need to make sure that the CI runs 1.7.0. So probably just bumping the required min version project-wide will resolve this at once.

File: examples/pytorch/text-classification/run_glue.py
Comment (bhavitvyamalik) [reply examples/pytorch/text-classification/run_glue.py:397]: Got the problem! Here, ds is not `Dataset` object but `DatasetDict` object. 
If you try `PYTHONPATH=src python -c 'from datasets import load_dataset; ds=load_dataset("glue", "cola"); ds["train"].map(desc="")'` that would work. I think I'll need to raise another PR for adding `desc` to DatasetDict objects as well [here](https://github.com/huggingface/datasets/blob/e08362256fb157c0b3038437fc0d7a0bbb50de5c/src/datasets/dataset_dict.py#L410)

Comment (stas00) [reply examples/pytorch/text-classification/run_glue.py:397]: Awesome.

hope you manage to get the fix into 1.7.0 - otherwise we will have to wait until the next release.

Comment (bhavitvyamalik) [reply examples/pytorch/text-classification/run_glue.py:397]: I've just raised the PR but 1.7.0 was released yesterday only :/

Comment (stas00) [reply examples/pytorch/text-classification/run_glue.py:397]: that will be a problem. since there is no way to compare the git version.

Once datasets has merged your additional fix, let's come back here and set this PR to be merged once the required version is there. it will be easy to ensure because the CI will fail until the new version will be available.

it's also very possible that they will be making hotfixes, as is often the case with minor version bumps. So it might be sooner than we think.

Comment (bhavitvyamalik) [reply examples/pytorch/text-classification/run_glue.py:397]: Sure, we can target a patch release too. Will let you know when that happens!

**Comment by bhavitvyamalik** [comment]
@stas00, Datasets 1.8.0 was released yesterday and now all tests are passing in this PR. Please let me know if it looks good or if any changes are required so that I can update the rest of them as well.

**Comment by stas00** [comment]
Thank you for the heads up, @bhavitvyamalik!

The code now works:
```
$ python run_glue.py   --model_name_or_path bert-base-cased   --task_name $TASK_NAME   --do_train   --do_eval   --max_seq_length 128   --per_device_train_batch_size 32   --learning_rate 2e-5   --num_train_epochs 3   --output_dir output_dir
[...]
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
Running tokenizer on dataset: 100%|████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 22.54ba/s]
Running tokenizer on dataset: 100%|████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 47.85ba/s]
Running tokenizer on dataset: 100%|████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:00<00:00, 14.15ba/s]
06/09
```

So what is missing is the version checking. normally we have all the versions setup in `setup.py` and its autogenerated table, and then we just need to do `dep_version_check("datasets")` but here this higher version is only needed for these few scripts. so one way is to code it explicitly:
```
from packaging import version
if version.parse(datasets.__version__) < version.parse("1.8.0")
    raise ValueError("....")
```


But at the same time is there any reason why we can't just bump transformers's library-wide dependency to `datasets==1.8.0`?

**Comment by stas00** [comment]
> @lhoestq can I work on check_min_version feature in datasets?

The problem with this one is that it will require a certain version of `datasets` to even work, so a generic mechanism is probably better.

but I forgot I already coded a wrapper :)
```
from transformers.utils.versions import require_version_
require_version("datasets>=1.8.0", "this is important to have for whatever reason")
```

**Comment by stas00** [comment]
Actually, the better way is to use `require_version_examples` and properly update the requirement in the requirement files, but I see the last reshuffle broke it all.

**Review by stas00 — APPROVED** [review]
This is good now! Thank you for all the fixups, @bhavitvyamalik

**Review by sgugger — APPROVED** [review]
Thanks for adding those!

**Comment by bhavitvyamalik** [comment]
@sgugger, I was yet to replicate this for other examples. I'll raise another PR for that!

**Comment by stas00** [comment]
I like the approach of making the first PR with just 1 or a few examples with a commitment to extend it to other examples in a separate PR (by creating an Issue that will track the replication).

The problem with tackling more than one example is that it puts a huge onus on the developer and reviewers to do much more work if things aren't right right away. But with one example, ideas are bounced around, tried, applied, verified, merged - then if all is good it's now easy to replicate to many more examples.

This is IMHO, of course.

**Pull request merged**
